server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/wrr/get/TransportGetWRRWeightsAction.java

```diff
@@ -0,0 +1,140 @@
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Transport action for getting weights for weighted round-robin search routing policy
+ *
+ * @opensearch.internal
+ */
+public class TransportGetWRRWeightsAction extends TransportClusterManagerNodeReadAction<
+    ClusterGetWRRWeightsRequest,
+    ClusterGetWRRWeightsResponse> {
+    private static final Logger logger = LogManager.getLogger(TransportGetWRRWeightsAction.class);
+    private volatile List<String> awarenessAttributes;
+
+    @Inject
+    public TransportGetWRRWeightsAction(
+        Settings settings,
+        ClusterSettings clusterSettings,
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(
+            ClusterGetWRRWeightsAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            ClusterGetWRRWeightsRequest::new,
+            indexNameExpressionResolver
+        );
+        this.awarenessAttributes = AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING.get(settings);
+        clusterSettings.addSettingsUpdateConsumer(
+            AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING,
+            this::setAwarenessAttributes
+        );
+    }
+
+    List<String> getAwarenessAttributes() {
+        return awarenessAttributes;
+    }
+
+    private void setAwarenessAttributes(List<String> awarenessAttributes) {
+        this.awarenessAttributes = awarenessAttributes;
+    }
+
+    @Override
+    protected String executor() {
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected ClusterGetWRRWeightsResponse read(StreamInput in) throws IOException {
+        return new ClusterGetWRRWeightsResponse(in);
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(ClusterGetWRRWeightsRequest request, ClusterState state) {
+        return null;
+    }
+
+    @Override
+    protected void clusterManagerOperation(
+        final ClusterGetWRRWeightsRequest request,
+        ClusterState state,
+        final ActionListener<ClusterGetWRRWeightsResponse> listener
+    ) throws IOException {
+        verifyAwarenessAttribute(request.getAwarenessAttribute());
+        Metadata metadata = state.metadata();
+        WeightedRoundRobinRoutingMetadata weightedRoundRobinMetadata = metadata.custom(WeightedRoundRobinRoutingMetadata.TYPE);
+        ClusterGetWRRWeightsResponse clusterGetWRRWeightsResponse = new ClusterGetWRRWeightsResponse();
+        String weight = null;
+        if (weightedRoundRobinMetadata != null && weightedRoundRobinMetadata.getWrrWeight() != null) {
+            WRRWeights wrrWeights = weightedRoundRobinMetadata.getWrrWeight();
+            if (request.local()) {
```

Comment:
`local` request should't need to be sent to cluster manager .

Reply:
Makes sense. I can add a check for cluster manager node

Reply:
Added cluster manager node check